philip-services-mcq.md

```diff
@@ -0,0 +1,110 @@
+<!--
+	Correct answer is 3.
+
+	1 - Similar to 4 below with the caveat that `Http` will not be passed into the constructor to begin with. Providers may only be injected via a typed constructor parameter (or @Inject token).
+
+  4 - While it is possible to instantiate classes manually, (and in some simple, stateless services you may actually get the behaviour you're expecting), manually instantiated classes lose the benefit of Angular's DI (their dependencies are no longer able to be dependency injected) and they require you to manually construct and manage a dependency tree yourself. Additionally,you're overriding the default singleton nature of providers in Angular which may lead to unexpected behaviour.
+
+	2 - It's possible that when students learn that dependencies are injected into a class constructor based on its type, they may expect this type reflection to also apply to class members.
+-->
+What is the correct way to inject a service into a class?
+
+1.
+    ```typescript
+    class MyComponent {
+      constructor(Http) {
+        this.http = new Http();
+      }
+    }
+    ```
+1.
+    ```typescript
+    class MyComponent {
+      private http: Http;
+    }
+    ```
+1.
+    ```typescript
+    class MyComponent {
+      constructor(http: Http) {
```

Comment:
Will learners have the same question that you did? Or can this be a question in its own right?  If it took you two looks...